pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -730,144 +1044,147 @@ protected BrokerResponse doHandleRequest(long requestId, String query, SqlNodeAn
       _brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.BROKER_RESPONSES_WITH_UNAVAILABLE_SEGMENTS, 1);
     }
 
-    if (offlineBrokerRequest == null && realtimeBrokerRequest == null) {
-      if (!exceptions.isEmpty()) {
-        ProcessingException firstException = exceptions.get(0);
-        String logTail = exceptions.size() > 1 ? (exceptions.size()) + " exceptions found. Logging only the first one"
-            : "1 exception found";
-        LOGGER.info("No server found for request {}: {}. {}", requestId, query, logTail, firstException);
-        _brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.NO_SERVER_FOUND_EXCEPTIONS, 1);
-        return new BrokerResponseNative(exceptions);
-      } else {
-        // When all segments have been pruned, we can just return an empty response.
-        return getEmptyBrokerOnlyResponse(pinotQuery, requestContext, tableName, requesterIdentity, schema, query,
-            database);
+    return new Requests(offlineRoutingTable, realtimeRoutingTable, offlineBrokerRequest, realtimeBrokerRequest,
+        numPrunedSegmentsTotal, exceptions);
+  }
+
+  private BrokerResponse handleLogicalTableQuery(long requestId, String query, RequesterIdentity requesterIdentity,
+      RequestContext requestContext, HttpHeaders httpHeaders, AccessControl accessControl, PinotQuery pinotQuery,
+      PinotQuery serverPinotQuery, LogicalTable logicalTable, String database, BrokerRequest brokerRequest,
+      BrokerRequest serverBrokerRequest)
+      throws Exception {
+    long authStartTime = System.nanoTime();
+    boolean ignoreCase = _tableCache.isIgnoreCase();
+    Schema schema = _tableCache.getSchema(logicalTable.getPhysicalTableNames().get(0));
+
+    List<LogicalTableRequests> brokerRequests = new ArrayList<>(logicalTable.getPhysicalTableNames().size());
+    long routingStartTimeNs = System.nanoTime();
+    for (String tableName : logicalTable.getPhysicalTableNames()) {
+      try {
+        LogicalTableUtils.HybridTableName hybridTableName =
+            LogicalTableUtils.getHybridTableName(tableName, _routingManager);
```

Comment:
The `tableName` here should always be `tableNameWithType`, and we should not construct hybrid table from it. Say I have `t_OFFLINE` and `t_REALTIME`, and in my logical table I only want to query `t_OFFLINE`, the current logic will also query `t_REALTIME` which is wrong